modules/vmseries/variables.tf

```diff
@@ -17,14 +17,21 @@ variable "network_interfaces" {
   description = <<-EOF
   List of the network interface specifications.
   Available options:
-  - `subnetwork`             - (Required|string) Self-link of a subnetwork to create interface in.
-  - `private_ip_name`        - (Optional|string) Name for a private address to reserve.
-  - `private_ip`             - (Optional|string) Private address to reserve.
-  - `create_public_ip`       - (Optional|boolean) Whether to reserve public IP for the interface. Ignored if `public_ip` is provided. Defaults to 'false'.
-  - `public_ip_name`         - (Optional|string) Name for a public address to reserve.
-  - `public_ip`              - (Optional|string) Existing public IP to use.
-  - `public_ptr_domain_name` - (Optional|string) Existing public PTR name to use.
-  - `alias_ip_ranges`        - (Optional|list) List of objects that define additional IP ranges for an interface, as specified [here](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#ip_cidr_range)
+  - `subnetwork`                  - (Required|string) Self-link of a subnetwork to create interface in.
+  - `stack_type`                  - (Optional|string) IP stack to use: IPV4_ONLY (default) or IPV4_IPV6.
+  - `private_ip_name`             - (Optional|string) Name for a private IPv4 address to reserve.
+  - `private_ip`                  - (Optional|string) Private IPv4 address to reserve.
+  - `create_public_ip`            - (Optional|boolean) Whether to reserve public IPv4 address for the interface. Ignored if `public_ip` is provided. Defaults to 'false'.
+  - `public_ip_name`              - (Optional|string) Name for a public IPv4 address to reserve.
+  - `public_ip`                   - (Optional|string) Existing public IPv4 address to use.
+  - `public_ptr_domain_name`      - (Optional|string) Existing public IPv4 address PTR name to use.
+  - `alias_ip_ranges`             - (Optional|list) List of objects that define additional IP ranges for an interface, as specified [here](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#ip_cidr_range)
+  - `create_public_ipv6`          - (Optional|boolean) Whether to reserve public IPv6 address for the interface. Ignored if `public_ipv6` is provided. Defaults to 'false'.
```

Comment:
Maybe it's worth to document that public and private ipv6 are mutually exclusive?

Reply:
This is more of a VPC-module-level question. It should be pretty obvious due to `ipv6_access_type` parameter:
```
    subnetworks = {
      fw-spoke1-sub = {
        name              = "fw-spoke1-sub"
        create_subnetwork = true
        stack_type        = "IPV4_IPV6"
        ip_cidr_range     = "192.168.1.0/28"
        ipv6_access_type  = "INTERNAL"
        region            = "us-east1"
      }
    }
```

I'm not sure that VM-Series module is a right place for describing GCP IPv6 internals.

Reply:
I'd say it's specific to this module as well, having it documented that it's either one or the other and the subnet has to have corresponding setting may be useful for users.